### ISSUE TITLE
Controlled multiselect - scroll position reset when option is selected

### ISSUE THREAD

**Issue by thedistricts** [benbowes/react-responsive-select#60]
Selecting an option in a controlled multiselect resets the menu's scroll position

**Demo**
[https://codepen.io/thedistricts/pen/gvaXwr](https://codepen.io/thedistricts/pen/gvaXwr)

1) Open Multiselect 
2) Select 'Test5'
3) Notice the scroll position has returned to the top of the menu 

Expected behaviour: Persist scroll position between selecting options

**Comment by benbowes** [comment]
Cheers for the Codepen @thedistricts!

I never did set up a scroll-to-selected-option for multi-select. Seems like a good time to do that.

I think it should be scroll to the first selected option in `selectedValues` if the multi-select has not been altered, or scroll to the last selected option in `selectedValues` if it has been altered.

**Comment by benbowes** [comment]
Hi @thedistricts 

I can see that fixing this particular issue would be fairly easy, but can also see that going forward some other internals are going to have to change to cater for selecting the first item when you "go and come back to" the component.

Would you like to see this particular issue sorted quickly or are you in a position to wait a week?

The only difference from my end is 2 publishes versus 1 publish

**Comment by thedistricts** [comment]
Thanks @benbowes - A week should be fine if that works for you. Much appreciated! I've had fork and implement a few small tweaks to get the flexibly I needed for a project. I'll note the suggestions in an issue, see if you think it's worth including in your main component.

**Comment by benbowes** [comment]
Nice one!

I'm thinking It would be a good idea to separate these two https://github.com/benbowes/react-responsive-select/blob/master/src/reducers/reducer.js#L48-L49 so that they have there own state manipulations.

Also... the quick fix I mentioned above was to changing this line:
https://github.com/benbowes/react-responsive-select/blob/master/src/reducers/reducer.js#L74

This
```
nextPotentialSelectionIndex: initialSelectedIndex,
```
To this
```
nextPotentialSelectionIndex: state.nextPotentialSelectionIndex || initialSelectedIndex,
```

The longer fix alters a more of the internal properties. I haven't had time to get into it yet. But will :)

**Comment by benbowes** [comment]
I'd dig hearing those suggestions

**Comment by benbowes** [comment]
Released 3.1.4 with https://github.com/benbowes/react-responsive-select/pull/63
